### PR TITLE
Merge PatternsCS to Patterns

### PR DESCRIPTION
**Pull request by He-Pin**
refs:https://github.com/akka/akka/issues/26000

### CODE REVIEW

File: akka-actor/src/main/scala/akka/pattern/Patterns.scala [hunk 1]
```diff
@@ -359,6 +510,7 @@ object PatternsCS {
    *   f.thenRun(result -> nextActor.tell(new EnrichedResult(request, result)));
    * }}}
    */
+  @deprecated("Use Patterns.ask instead.", since = "2.5.19")
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/pattern/Patterns.scala]: Should them be deprecated or not?

**Review by patriknw — APPROVED** [review]
LGTM, perhaps with more deprecations?

**Comment by hepin1989** [comment]
@patriknw can't ,because there are `timeInMillins:Long ,timeUnit:TimeUnit` like thing...

**Comment by hepin1989** [comment]
image
The methods which accept `Long` are there, so seems like no way to deprecate them,otherwise deprecate these methods too.

**Comment by patriknw** [comment]
> timeInMillins:Long ,timeUnit:TimeUnit

I think that one can be deprecated in favor of using java.time.Duration

**Comment by hepin1989** [comment]
@patriknw Okay，that would be great, then how about those in Scala API?

**Comment by hepin1989** [comment]
@patriknw I have updated the PR and deprecated the PatternsCS.

**Review by patriknw — APPROVED** [review]
LGTM, again

File: akka-actor/src/main/scala/akka/pattern/Patterns.scala [hunk 2]
```diff
@@ -69,6 +69,36 @@ object Patterns {
...
+   * the containing actor’s object, i.e. do not call methods or access mutable state
+   * on the enclosing actor from within the callback. This would break the actor
+   * encapsulation and may introduce synchronization bugs and race conditions because
+   * the callback will be scheduled concurrently to the enclosing actor. Unfortunately
+   * there is not yet a way to detect these illegal accesses at compile time.
+   *
+   * <b>Recommended usage:</b>
+   *
+   * {{{
+   *   final CompletionStage<Object> f = PatternsCS.ask(worker, request, duration);
```
Comment (chbatey) [inline akka-actor/src/main/scala/akka/pattern/Patterns.scala]: Patterns.ask or remove the code example in the comment as they often end up not compiling

File: akka-actor/src/main/scala/akka/pattern/Patterns.scala [hunk 3]
```diff
@@ -83,6 +113,24 @@ object Patterns {
   def askWithReplyTo(actor: ActorRef, messageFactory: japi.Function[ActorRef, Any], timeout: Timeout): Future[AnyRef] =
     extended.ask(actor, messageFactory.apply _)(timeout).asInstanceOf[Future[AnyRef]]
 
+  /**
+   * A variation of ask which allows to implement "replyTo" pattern by including
+   * sender reference in message.
+   *
+   * {{{
+   * final CompletionStage<Object> f = PatternsCS.askWithReplyTo(
```
Comment (chbatey) [inline akka-actor/src/main/scala/akka/pattern/Patterns.scala]: Same

File: akka-actor/src/main/scala/akka/pattern/Patterns.scala [hunk 4]
```diff
@@ -165,6 +213,36 @@ object Patterns {
...
+   * the containing actor’s object, i.e. do not call methods or access mutable state
+   * on the enclosing actor from within the callback. This would break the actor
+   * encapsulation and may introduce synchronization bugs and race conditions because
+   * the callback will be scheduled concurrently to the enclosing actor. Unfortunately
+   * there is not yet a way to detect these illegal accesses at compile time.
+   *
+   * <b>Recommended usage:</b>
+   *
+   * {{{
+   *   final CompletionStage<Object> f = PatternsCS.ask(selection, request, duration);
```
Comment (chbatey) [inline akka-actor/src/main/scala/akka/pattern/Patterns.scala]: Same

**Review by chbatey — APPROVED** [review]
Check all the code in comments then LGTM

File: akka-actor/src/main/scala/akka/pattern/Patterns.scala (hunk 2 above)
Comment (patriknw) [reply akka-actor/src/main/scala/akka/pattern/Patterns.scala]: I'll follow up and fix this comments. Want this to be merged for release validation now.

**Pull request merged**